compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -923,12 +962,22 @@ trait Implicits { self: Typer =>
       val ref = cand.ref
       var generated: Tree = tpd.ref(ref).withPos(pos.startPos)
       val locked = ctx.typerState.ownedVars
-      if (!argument.isEmpty)
-        generated = typedUnadapted(
-          untpd.Apply(untpd.TypedSplice(generated), untpd.TypedSplice(argument) :: Nil),
-          pt, locked)
-      val generated1 = adapt(generated, pt, locked)
-
+      val generated1 =
+        if (argument.isEmpty)
+          adapt(generated, pt, locked)
+        else {
+          val untpdGenerated = untpd.TypedSplice(generated)
+          if (cand.isConversion)
+            typed(
+              untpd.Apply(untpdGenerated, untpd.TypedSplice(argument) :: Nil),
+              pt, locked)
+          else {
+            assert(cand.isExtension)
+            val SelectionProto(name: TermName, mbrType, _, _) = pt
+            extMethodApply(untpd.Select(untpdGenerated, name), argument, mbrType)
+          }
+          // TODO disambiguate if candidate can be an extension or a conversion
```

Comment:
Should an issue be opened for this todo ?

Reply:
I pushed a commit that addresses the TODO